Add a couple isset to module.

jonathangreen opened the pull request:
## JIRA Ticket

https://jira.duraspace.org/browse/ISLANDORA-2015

## What does this Pull Request do?

Adds some `isset` checks around variables in the template and preprocess functions, so we don't throw warnings when derivatives have not been created yet.

## What's new?

Should be no change in functionality, just some more checks.

## How should this be tested?

* Disable derivatives in the video module
* Ingest video
* See PHP warnings (make sure warnings are enabled).

## Interested parties
@Islandora/7-x-1-x-committers

whikloj commented on islandora_videojs.module:
```diff
@@ -64,9 +66,12 @@ function template_preprocess_islandora_videojs(array &$variables) {
     $variables['sources'][] = array('url' => $params['url'], 'mime' => $params['mime']);
   }
+  else {
+    $variables['sources'] = array();
```
So this works nicely to remove the warnings, but you are left with what appears to be a working player but nothing plays. Maybe we should add a warning like.
```
drupal_set_message("The video file is missing from this object. Please contact the site administrator.", "warning");
```
into this else statement?

DiegoPino commented on islandora_videojs.module:
```diff
@@ -64,9 +66,12 @@ function template_preprocess_islandora_videojs(array &$variables) {
     $variables['sources'][] = array('url' => $params['url'], 'mime' => $params['mime']);
   }
+  else {
+    $variables['sources'] = array();
```
I would avoid the `else` at line 72  at all and just initialize `$variables['sources'] = array();` at the beginning of the function.  We could also, simply drop adding the libraries if sources is empty at all, both in the preprocess and the tpl. file? and just display the poster frame if set? Plus the message @whikloj correctly suggests which means i also support adding that message. Only, minor note, i would discourage any messages that encourage contacting the site administrator, which is for sure someone unknown for an anonymous visitor.

whikloj replied:
Fair enough, I'm not tied to that text. I don't mind showing the player, I just want people to be aware that there is nothing to play on this object.

rosiel replied:
I agree with Diego (as a site administrator) on his minor note. May I propose something along the lines of "Video file not found."?

jonathangreen commented:
I don't like the idea of using a `drupal_set_message` to convey this information, as its hard to override and what it means not to have a video file is a site specific thing. In my instance it very likely means that the video is being processed by the derivative server, and I would like to be able to message this at the theme layer. 

I implemented @DiegoPino suggestion not to use an else statement. I added a message to the default template that prints a message if there are no sources available, does that work for everyone?

whikloj commented:
I'm good with the changes. @rosiel any concerns?

DiegoPino commented:
@whikloj I think @rosiel is still on vacations. If you are all good with waiting for her, that would be nice and cool.

rosiel commented:
Can't test the code, but it looks good!
